server/controllers/searchRecipe.js

```diff
@@ -0,0 +1,155 @@
+        ],
+        order: [
+          ['upvotes', 'DESC']
+        ]
+      })
+      .then((foundRecipes) => {
+        if (!foundRecipes) {
+          return res.status(404).json({
+            success: true,
+            message: 'No Stored Recipes found',
+          });
+        }
+
+        return res.status(201).json({
+          success: true,
+          data: foundRecipes });
+      })
+      .catch(() => res.status(503).json({
+        success: false,
+        message: 'Unable to fetch recipes' }));
+
+    return this;
+  }
+
+  /**
+   * Search for recipe by Recipe name, Ingredients or Name of User
+   *
+   * @param {object} req - HTTP Request
+   * @param {object} res - HTTP Response
+   * @returns {object} Class instance
+   * @memberof Search
+   */
+  searchAll(req, res) {
+    let results;
+    const searchTerm = req.query.search;
+
+    recipe
+      .findAll({
+        where: {
+          $or: [
+            { name: {
+              $iLike: `%${searchTerm}%` }
+            },
+            { ingredients: {
+              $iLike: `%${searchTerm}%` }
+            }
+          ]
+        },
+        include: [
+          { model: models.User, attributes: ['name', 'updatedAt'] }
+        ]
+      })
+      .then((foundRecipes) => {
+        results = foundRecipes.slice(0);
+      })
+      .then(() => {
+        user
+          .findAll({
+            attributes: ['name'],
+            where: {
+              $or: [
+                { name: {
+                  $iLike: `%${searchTerm}%` }
+                },
+                { username: {
+                  $iLike: searchTerm }
+                },
+                { email: {
+                  $iLike: searchTerm }
+                },
+              ]
+            },
+            include: [
+              { model: models.Recipe }
+            ]
+          })
+          .then(data => res.status(201).json({
+            success: true,
+            data: results.concat(data) }));
+      })
+      .catch(() => res.status(503).json({
+        success: false,
+        message: 'Unable to search recipes' }));
+
+    return this;
+  }
+
+  /**
+   * Fetch list of recipes based ingredients supplied
+   *
+   * @param {object} req - HTTP Request
+   * @param {object} res - HTTP Response
+   * @returns {object} Class instance
+   * @memberof Search
+   */
+  searchByIngredients(req, res) {
+    const ingredients = req.query.ingredients.split(' ');
+
+    const queryClause = ingredients.map(item => ({
+      ingredients: { $iLike: `%${item}%` } }));
+
+    recipe
+      .findAll({
+        where: {
+          $or: queryClause
+        },
+        include: [
+          { model: models.User, attributes: ['name', 'updatedAt'] }
+        ]
+      })
+      .then((foundRecipes) => {
+        if (!foundRecipes) {
+          return res.status(404).json({
+            success: true,
+            message: 'Nothing found',
+          });
+        }
```

Comment:
`else` statement missing

Reply:
Okay, but the thing is eslint warns that it's unnecessary to add an else statement to an if block that has a return statement within it.